### PR TITLE
Github Workflow Replacement for Jenkins Job, beam_PreCommit_GoPortable (Cron, Commit, Phrase)

### PR DESCRIPTION
**Pull request by magicgoody**
Similar workflow migration PR as reference: https://github.com/apache/beam/pull/27042

PR migrates Jenkins Jobs to Github Actions Workflow ".github/workflows/beam_PreCommit_GoPortable.yml":

https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_GoPortable_Commit/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_GoPortable_Cron/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_GoPortable_Phrase/
The code has been tested on Self-hosted GKE runners ubuntu 20.04, and on Self-hosted local ubuntu 20.04 machine.

The tests was made by creating pull requests with changes in trigger paths, by adding a comment to the PR, by creating a push.

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_GoPortable.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    timeout-minutes: 120
+    if: |
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' || 
+      github.event.comment.body == 'Run GoPortable PreCommit' || 
+      github.event_name == 'schedule'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    name: beam_PreCommit_GoPortable
+    steps:
+      - uses: actions/checkout@v3
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_GoPortable.yml:63]: Please add the `name` attribute for this step

**Review by andreydevyatkin — COMMENTED** [review]
Left a minor comment, overall LGTM

File: .github/workflows/beam_PreCommit_GoPortable.yml (hunk 1 above)
Comment (magicgoody) [reply .github/workflows/beam_PreCommit_GoPortable.yml]: Fixed, thanks

**Review by andreydevyatkin — APPROVED** [review]
LGTM

File: .github/workflows/beam_PreCommit_GoPortable.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Rerun on comment
+        if: github.event.comment.body == 'Run GoPortable PreCommit'
+        uses: ./.github/actions/rerun-job-action
+        with:
+          pull_request_url: ${{ github.event.issue.pull_request.url }}
+          github_repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ github.job }}
```
Comment (damccorm) [inline .github/workflows/beam_PreCommit_GoPortable.yml:75]: Should we make this `beam_PreCommit_GoPortable (Run GoPortable PreCommit)`?

**Review by damccorm — APPROVED** [review]
LGTM

**Pull request merged**